File: native/native.tests/testData/framework/objcexport/kt43599.swift
/*
 * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the LICENSE file.
 */

import Kt

private func testPropertyWithPrivateSetter() throws {
    try assertEquals(actual: KT43599().memberProperty, expected: "memberProperty")
    try assertEquals(actual: KT43599().extensionProperty, expected: "extensionProperty")
    try assertEquals(actual: Kt43599Kt.topLevelProperty, expected: "topLevelProperty")

    // Checking the reported case too:
    Kt43599Kt.setTopLevelLateinitProperty(value: "topLevelLateinitProperty")
    try assertEquals(actual: Kt43599Kt.topLevelLateinitProperty, expected: "topLevelLateinitProperty")
}

class Kt43599Tests : SimpleTestProvider {
    override init() {
        super.init()

        test("TestPropertyWithPrivateSetter", testPropertyWithPrivateSetter)
    }
}

File: native/native.tests/testData/framework/objcexport/kt43599.kt
/*
 * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the LICENSE file.
 */

package kt43599

// Note: this test relies on two-stage compilation.

class KT43599 {
    var memberProperty = "memberProperty"
        private set
}

var KT43599.extensionProperty
    get() = "extensionProperty"
    private set(value) { TODO() }

var topLevelProperty
    get() = "topLevelProperty"
    private set(value) { TODO() }

lateinit var topLevelLateinitProperty: String
    private set

fun setTopLevelLateinitProperty(value: String) {
    topLevelLateinitProperty = value
}
